pkg/log/logger.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 Outreach Corporation. All Rights Reserved.
+
+// Description: Implements a compatible API for the olog package and log package.
+
+package log
+
+import (
+	"context"
+	"io"
+	"log/slog"
+
+	"github.com/getoutreach/gobox/pkg/olog"
+	"github.com/sirupsen/logrus"
+)
+
+// Logger is the compatible log interface
+type Logger struct {
+	logrusLogger *logrus.Logger
+	ologLogger   *slog.Logger
+	useLogrus    bool
+}
+
+func New(useLogrus bool) *Logger {
+	if useLogrus {
+		l := logrus.New()
+		return &Logger{logrusLogger: l, useLogrus: true}
+	}
+	l := olog.New()
+	return &Logger{ologLogger: l, useLogrus: false}
+}
+
+func (log *Logger) Debug(ctx context.Context, message string, m Marshaler) {
```

Comment:
question: The current use of `pkg/log`'s logging methods allow multiple `m Marshaler`, and it's used a lot that way in services, e.g.

```go
log.Error(ctx, "Something bad happened", err, log.F{"user.email", "hello@example.com"})
```

How will this need to be migrated over?